### DISCUSSION TITLE
Help with rpi4 mod build and osc remote

### DISCUSSION THREAD

**Discussion by protectorateindustrial** [DISTRHO/Cardinal#593] [General]
Hi!

I`ve been trying to build mod lv2 versions for the raspberry pi 4. And I was actually able to build the plugins using a mainly the headless and mod_build flags, but osc remote is disabled due to binaries being plugins. What can I do to fix this?

Also, I see that the automated builds here use the mod-plugin-builder for the builds of mod binaries. Could I use the mod-plugin-builder toolchain for rpi4 to build Cardinal Mod lv2? Would there be a performance benefit of doing so? How would I go about doing this?

**Comment by falkTX** [comment]
you need to build it yourself to get this, there are far too many different ARM systems that due to their slower-than-usual CPUs do not really work well with a generic binary.

and yes, the plugin versions do not have the remote control thing since it would easily break when running more than 1 instance, also gets in a weird situation regarding automation of parameters.
so only the standalones have the feature, this being by design.

**Reply by protectorateindustrial** [reply]
Ok :D thanks for the quick answer. I've been able to build the mod plugins, the mod gui works and theres sound and I'm able to load presets in mod-ui. How can I build a versions that have the osc remote enabled? The binaries for moddwarf still have the osc remote enabled and also work in my system, how can I do that with my own build? Do I need to use specific build flags or something?

**Reply by falkTX** [reply]
the workflow file describes the build https://github.com/DISTRHO/Cardinal/blob/main/.github/workflows/build.yml#L256

but I think it is just an older version while that was still enabled.
or maybe headless builds always have it on? tbh I kinda forgot now

**Reply by protectorateindustrial** [reply]
I have both headless and mod_build set to true, I copied the make command from moddwarf build log, without the custom toolchain stuff, but the remote isnt working.

The mod dwarf build I tested is 23.09, so its not an older one, and you'd want to have the osc remote enabled on the mod build anyway, since you cant edit the plugin without the osc remote functionality.

**Reply by protectorateindustrial** [reply]
It would be good to be able to switch the osc remote on for plugin builds. There are a couple of things that need this. The mod buld for one, but I'm also thinking of compiling cardinal for Elk OS, and that would need to be a headless plugin and have the osc remote enabled. I did test the moddwarf lv2 on Elk OS and the it ran fine.

**Reply by protectorateindustrial** [reply]
I managed to build the mod builds for rpi4 by simply adding the rpi4 to the build rule list in dfp/makefile.base.mk. the mod-plugin-builder cross tool seemed to work with the build. performance is slightly better and cpu is much more stabile on the rpi4 compared to the moddwarf binary.
